cdep/src/main/java/io/cdep/cdep/generator/CMakeGenerator.java

```diff
@@ -85,12 +96,31 @@ public void generate() throws IOException {
   }
 
   @NotNull
-  public String create() {
+  String create() {
     append("# GENERATED FILE. DO NOT EDIT.\n");
     append(readCmakeLibraryFunctions());
     for (StatementExpression findFunction : table.findFunctions.values()) {
       indent = 0;
       visit(findFunction);
+      // The following cover the case that there's a header only dependency with header only dependency along with maintaining old behavior.
+      // A better solution would be to create a dependency type so we'd know in the in the visit function if the librarypath will never get called in subsequent visits.
+      if(insertingIndex > 0) {
```

Comment:
Please match the convention of using { } even for single line. Here and elsewhere